gc/base/MemoryPoolLargeObjects.cpp

```diff
@@ -298,28 +297,11 @@ MM_MemoryPoolLargeObjects::resizeLOA(MM_EnvironmentBase* env)
 				oldLOARatio = _currentLOARatio;
 
 				/* Does this leave a reasonable sized LOA ? */
-				if (!isSizeEnoughForLOA(env, (_loaSize > spaceDelta) ? (_loaSize - spaceDelta) : 0)) {
-					/* No.. make LOA empty as not even big enough for one free chunk of LOA */
-					spaceDelta = _loaSize;
-					_soaSize += spaceDelta;
-					_loaSize = 0;
-					_currentLOABase = LOA_EMPTY;
-					_currentLOARatio = 0;
-				} else {
-					_soaSize += spaceDelta;
-					_loaSize -= spaceDelta;
-					_currentLOABase = newLOABase;
-					_currentLOARatio = ((double)_loaSize) / (_loaSize + _soaSize);
-
-					/* Rounding during float operations may result in a new LOA ratio less than
-					 * minimum so fix up if necessary
-					 */
-					if (_currentLOARatio < _extensions->largeObjectAreaMinimumRatio) {
-						_currentLOARatio = _extensions->largeObjectAreaMinimumRatio;
-					}
-					assume0(0 != _currentLOARatio);
+				newLOAsize = (_loaSize > spaceDelta) ? (_loaSize - spaceDelta) : 0;
+				uintptr_t oldLOAsize = _loaSize;
+				if (!isSizeEnoughForLOA(env, (_loaSize + _soaSize), newLOAsize, 0, newLOABase)) {
```

Comment:
Let's not pass total old area size as an argument. we can fetch it inside the helper.
we can assert here, prior to calling that 

_memorySubSpace->getActiveMemorySize() == _loaSize + _soaSize